src/components/ScanOrder.js

```diff
@@ -77,50 +82,35 @@ const ScanOrder = (props) => {
   };
 
   useEffect(() => {
-    retrieveStatuses();
-  }, []);
-
-  const dynamicSort = (property) => {
-    let sortOrder = 1;
-
-    if (property[0] === "-") {
-      sortOrder = -1;
-      property = property.substr(1);
+    if (statuses.length === 0) {
+      retrieveStatuses();
     }
-
-    return function (a, b) {
-      let result =
-        a[property] < b[property] ? -1 : a[property] > b[property] ? 1 : 0;
-      return result * sortOrder;
-    };
-  };
+  }, [statuses]);
 
   let nextDesc = "";
   let nextId = null;
   let nextSeq = null;
   let nextStatusFedOfficeCode = "";
+  let sortedStatuses = [];
 
-  if (STATUSES && order) {
-    STATUSES.sort(dynamicSort("sequence_num"));
+  if (statuses && order) {
+    sortedStatuses = numSort(statuses, "sequence_num", "asc");
 
     const currentSeq = order.status.sequence_num;
 
-    for (let i = 0; i < STATUSES.length - 1; i++) {
-      if (STATUSES[i].sequence_num > currentSeq) {
-        nextDesc = STATUSES[i].description;
-        nextId = STATUSES[i].id;
-        nextSeq = STATUSES[i].sequence_num;
-        nextStatusFedOfficeCode = STATUSES[i].status_federal_office_code;
+    for (let i = 0; i < sortedStatuses.length - 1; i++) {
+      if (sortedStatuses[i].sequence_num > currentSeq) {
+        nextDesc = sortedStatuses[i].description;
+        nextId = sortedStatuses[i].id;
+        nextSeq = sortedStatuses[i].sequence_num;
+        nextStatusFedOfficeCode = sortedStatuses[i].status_federal_office_code;
         break;
       }
-      if (i === STATUSES.length - 2) {
-        nextDesc = "FINAL";
-      }
     }
   }
 
   const handleUpdate = () => {
-    setOrder({
+    const upOrder = {
```

Comment:
Extremely nitpicky but I'd "waste" a few extra characters to call this "updatedOrder" for readability

Reply:
Makes sense, I'll do this in my next update as I don't want to risk having to reresolve conflicts if possible, but I will make a note to switch this as recommended